import "common/skill_common.proto";

package protos.common;

option java_package = "dass.server.gameserver.protos.common";
option java_outer_classname = "MonsterCommon";
option optimize_for = SPEED ;

//怪物
message Monster{
	optional int32	id = 1;			//关卡怪物ID(序号ID)
	optional int32  monsterId = 2;		//怪物ID(标识ID)
	optional string name = 3;		//怪物名称
	optional int32	atk = 4;		//怪物攻击力
	optional int32	def = 5;		//怪物防御
	optional float	atkInterval = 6;	//攻击速度
	optional int32	hp = 7;			//生命值
	optional int32  crit = 8;		//暴击值
	optional int32  hit = 9;		//命中
	optional int32	dodge = 10;		//回避
	optional int32  renew = 11;		//回复
	optional int32  minRound = 12;		//必杀技最少施放回合
	optional int32	maxRound = 13;		//必杀技最大施放回合
	optional int32	addHurtRate = 14;	//增伤比
	optional int32  cutHurtRate = 15;	//减伤比
	optional int32	thumb = 16;		//怪物物模型标识
	repeated Pos	posList = 17;		//坐标列表
	optional int32  zoom = 18;		//模型缩放比率
	optional int32  move1 = 19;		//方向1移动范围
	optional int32  move2 = 20;		//方向2移动范围
	optional int32  move3 = 21;		//方向3移动范围
	optional int32  move4 = 22;		//方向4移动范围
	optional int32  move5 = 23;		//方向5移动范围
	optional common.Skill skill1 = 24;	//普通攻击
	optional common.Skill skill2 = 25;	//特殊攻击
	optional common.Skill skill3 = 26;	//主动技能
	optional int32  foe = 28;		//控制类型
	optional int32	hasItem = 29;		//击杀后是否掉落道具
	optional int32  monsterType = 30;	//怪物类型
	optional float  moveSpeed = 31;		//移动速度
	optional int32  level = 32;		//怪物等级
	optional int32	quality = 33;		//怪物品质
	optional bool	isBoss = 34;		//是否
	optional int32  buff = 40;		//怪物附带特效(如：隐身、自爆等)

	optional bool   isCall = 35;		//是否召唤的怪物
	optional int32	initCost = 36;		//初始cost
	optional int32	addCost = 37;		//cost恢复速度(1/s)
	optional int32  maxCost = 39;		//cost上限
	optional int32  delay = 41;		//怪物延时出现时间
	optional int32  batch = 42;		//怪物批次
	optional int32  posX = 43;		
	optional int32  posY = 44;
	optional int32  maxHp = 45;		//最大生命值
}

message Pos{
	required int32 x = 1;
	required int32 y = 2;
}